crypto/ec/ec_pmeth.c

```diff
@@ -121,8 +124,23 @@ static int pkey_ec_sign(EVP_PKEY_CTX *ctx, unsigned char *sig, size_t *siglen,
     }
 
     type = (dctx->md != NULL) ? EVP_MD_type(dctx->md) : NID_sha1;
-
-    ret = ECDSA_sign(type, tbs, tbslen, sig, &sltmp, ec);
+	
+    switch (dctx->nonce_type) {
+    case EVP_PKEY_ECDSA_NONCE_RANDOM:
+        ret = ECDSA_sign(0, tbs, tbslen, sig, &sltmp, ec);
```

Comment:
why has the type been removed?

Reply:
if type>0 the ECDSA_sign_setup would consider it as the nid of the hash algorithm. Thus, produce undesiered deterministic signature

Reply:
ok - it didnt use the type before..